### PR TITLE
feat: save selection

### PR DESCRIPTION
**Pull request by FritzHoing**
a Modal has been created, that gets the link of the current Map.

For now:
Additional WMS-Layers are not included in that link. WIP.

### CODE REVIEW

**Review by Unraveler — CHANGES_REQUESTED** [review]
You should fix the warnings displayed here. Otherwise, it looks good to me.

File: src/components/SaveSelectModal/SaveSelectModal.tsx [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+  let map = useMap();
+
+  const closeModal = () => {
+    dispatch(hideSelect());
+  };
+
+  function onCopyClick() {
+    if (map) {
+
```
Comment (dnlkoch) [inline src/components/SaveSelectModal/SaveSelectModal.tsx:54]:
```suggestion
```

File: src/components/SaveSelectModal/SaveSelectModal.tsx [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+  const dispatch = useAppDispatch();
+  const { t } = useTranslation();
+
+  let map = useMap();
+
+  const closeModal = () => {
+    dispatch(hideSelect());
+  };
+
+  function onCopyClick() {
```
Comment (dnlkoch) [inline src/components/SaveSelectModal/SaveSelectModal.tsx:52]: Please convert to arrow function.

File: src/components/SaveSelectModal/SaveSelectModal.tsx [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+      if (success) {
+        message.info(t('SaveSelectModal.success'));
+      } else {
+        message.info(t('SaveSelectModal.failure'));
+      }
+    }
+  };
+
+  function link() {
```
Comment (dnlkoch) [inline src/components/SaveSelectModal/SaveSelectModal.tsx:66]: Please convert to arrow function.

File: src/components/SaveSelectModal/SaveSelectModal.tsx [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+        <Tooltip title={t('SaveSelectModal.tooltip')}>
+          <CopyOutlined onClick={onCopyClick} />
+        </Tooltip>
+      </div>
+      <div className="link">
+        <Input value={link()} readOnly />
+      </div>
+    </Modal>
+  );
+};
```
Comment (dnlkoch) [inline src/components/SaveSelectModal/SaveSelectModal.tsx:105]:
```suggestion
};

```

File: src/hooks/useQueryParams.ts [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0.txt
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
```
Comment (dnlkoch) [inline src/hooks/useQueryParams.ts:18]:
```suggestion
```

File: src/components/SaveSelectModal/SaveSelectModal.tsx [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+export default SaveSelectModal;
```
Comment (dnlkoch) [inline src/components/SaveSelectModal/SaveSelectModal.tsx:106]: I would prefer to have this component renamed to something like `PermalinkModal` or similar.

**Review by dnlkoch — CHANGES_REQUESTED** [review]
Thanks @FritzHoing!

I noted some minor remarks and as @Unraveler stated already, please check the linter warning.

File: src/components/PermalinkModal/index.tsx [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+        />
+        <Tooltip title={t('Permalink.copyTooltip')}>
+          <CopyOutlined onClick={onCopyClick} />
+        </Tooltip>
+      </div>
+    </div>
+  );
+};
+
+export default SharePanel;
```
Comment (dnlkoch) [inline src/components/PermalinkModal/index.tsx:99]: The file should be renamed accordingly.

File: src/components/PermalinkModal/index.tsx [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+
+  if (!map) {
+    return <></>;
+  }
+
+  const link = PermalinkUtil.getLink(map);
+
+  // TODO: these could be props
+  const mailSubject = 'Geoportal Raumordnung Kartenviewer';
+  const mailBody = `Hey,\r\nsieh dir an was ich im Geoportal Raumordnung Baden-Württemberg gefunden habe:\r\n${link}`;
```
Comment (dnlkoch) [inline src/components/PermalinkModal/index.tsx:44]: This should be adjusted.

File: src/components/PermalinkModal/index.tsx [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
+
+  function onTwitterClick() {
+    const twitterUrl = new URL('https://twitter.com/intent/tweet');
+    twitterUrl.searchParams.set('url', mailBody);
+    window.open(twitterUrl);
+  }
+
+  function onWhatsAppClick() {
+    const whatsAppUrl = new URL('http://wa.me');
```
Comment (dnlkoch) [inline src/components/PermalinkModal/index.tsx:53]:
```suggestion
    const whatsAppUrl = new URL('https://wa.me');
```

Untested.

File: src/i18n/translations.ts [hunk 10]
```diff
@@ -135,7 +150,8 @@ export default {
         featureInfo: 'Query map features',
         addWms: 'Add WMS',
         print: 'Export',
-        layertree: 'Maps'
+        layertree: 'Maps',
+        saveSelection: 'Save selection'
```
Comment (dnlkoch) [inline src/i18n/translations.ts:154]: Is this still used?

**Review by dnlkoch — CHANGES_REQUESTED** [review]

**Review by dnlkoch — APPROVED** [review]

**Pull request merged**